src/inputs/internal/SelectFieldBase.tsx

```diff
@@ -400,3 +425,23 @@ function getInputValue<O>(
     ? nothingSelectedText
     : "";
 }
+
+export function initializeOptions<O>(options: OptionsOrLoad<O>, unsetLabel: string | undefined): OptionsOrLoad<O> {
+  if (!unsetLabel) {
+    return options;
+  }
+
+  if (Array.isArray(options)) {
+    return getOptionsWithUnset(unsetLabel, options);
+  }
+
+  return { ...options, initial: getOptionsWithUnset(unsetLabel, options.initial) };
+}
+
+function getOptionsWithUnset<O>(unsetLabel: string, options: O[]): O[] {
+  return [{ id: undefined, name: unsetLabel } as unknown as O, ...options];
+}
+
+function isUnsetOption<O>(o: UnsetOption | O): o is UnsetOption {
```

Comment:
We could potentially replace the `{ id: undefined, name: unsetLabel }` hash with just at top-level `const` like:

```
const unsetOption = {};
```

And then `isUnsetOption` may not be necessary because we'd just do `o === unsetOption`?